### DISCUSSION TITLE
ContentPresenter.Styles sometime work, sometime not...

### DISCUSSION THREAD

**Discussion by dbriard** [AvaloniaUI/Avalonia#16915] [Q&A]
Hi,

In my theme, for many controls that can have Content, I am using ContentPresenter.Styles to restyle controls used as content.
But I have a few cases where that do not work and I can't figure out why.

For example:
```
<ContentPresenter.Styles>
  <Style Selector="ToggleSwitch">
        <Setter Property="Width" Value="56" />
    </Style>
</ContentPresenter.Styles>
```
In one of my Control, it work, same code at another place, it don't work.

Anyone have any idea?

**Comment by workgroupengineering** [comment]
Explore the visual tree with DevTools and find the Control where the style is not applied and see the style precedence. Remember to select `Show inactive`

![immagine](https://github.com/user-attachments/assets/b2062551-1270-410e-a21c-ff3d02466575)

**Reply by dbriard** [reply]
Found something...

I don't know why but it seems that it won't work when the ContentPresenter is binded on Content property (of ContentControl).
But it works for other properties that I create myself... they should be something with Content property.

Same with HeaderedControlControl, it doesn't work on Control and Header... 

I tried to add the code below to my own property and this is what broke the Styles... I don't know if it is normal or a bug?

```
private void HeaderChanged(AvaloniaPropertyChangedEventArgs e)
{
    if (e.OldValue is ILogical oldChild)
    {
        LogicalChildren.Remove(oldChild);
    }

    if (e.NewValue is ILogical newChild)
    {
        LogicalChildren.Add(newChild);
    }
}
```
